tracer/src/Datadog.Trace/Util/DbCommandCache.cs

```diff
@@ -39,8 +39,21 @@ internal static class DbCommandCache
 
         public static TagsCacheItem GetTagsFromDbCommand(IDbCommand command)
         {
-            var connectionString = command.Connection?.ConnectionString;
+            IDbConnection dbConnection = null;
+            try
+            {
+                dbConnection = command.Connection;
+            }
+            catch (NotSupportedException nsException)
+            {
+                Log.Warning(nsException, "Connection cannot be retrieved from the command.");
```

Comment:
Given this is an "expected" exception, I think we should make this a Debug, so as to not pollute the logs (as presumably it could happen with _every_ sql span in a customer's app if they're using `DbDataSource`)?

```suggestion
                Log.Debug(nsException, "Connection cannot be retrieved from the command.");
```